MediaWikiKit/MediaWikiKit/MWKLocationSearchResult.m

```diff
@@ -0,0 +1,60 @@
+
+#import "MWKLocationSearchResult.h"
+#import "NSURL+Extras.h"
+#import "NSString+Extras.h"
+
+@implementation MWKLocationSearchResult
+
++ (NSValueTransformer*)thumbnailURLJSONTransformer {
+    return [MTLValueTransformer transformerUsingForwardBlock:^id (NSString* value, BOOL* success, NSError* __autoreleasing* error) {
+        
+        return [NSURL wmf_optionalURLWithString:value];
+    }];
+}
+
++ (NSValueTransformer*)locationJSONTransformer {
+    return [MTLValueTransformer transformerUsingForwardBlock:^id (NSArray* value, BOOL* success, NSError* __autoreleasing* error) {
+        
+        NSDictionary* coords = [value firstObject];
+        NSNumber* lat = coords[@"lat"];
+        NSNumber* lon = coords[@"lon"];
+
+        if(!lat || !lon){
+            return nil;
+        }
+        
+        return [[CLLocation alloc] initWithLatitude:[lat doubleValue] longitude:[lon doubleValue]];
+    }];
+}
+
++ (NSValueTransformer*)wikidataDescriptionJSONTransformer {
+    return [MTLValueTransformer transformerUsingForwardBlock:^id (NSArray* value, BOOL* success, NSError* __autoreleasing* error) {
+        
+        NSString* description = [value firstObject];
+        return [description wmf_stringByCapitalizingFirstCharacter];
+    }];
+}
+
++ (NSValueTransformer*)distanceFromQueryCoordinatesJSONTransformer {
+    return [MTLValueTransformer transformerUsingForwardBlock:^id (NSArray* value, BOOL* success, NSError* __autoreleasing* error) {
+        
+        NSDictionary* coords = [value firstObject];
+        NSNumber* distance = coords[@"dist"];
+        return distance;
+    }];
+}
+
+
++ (NSDictionary*)JSONKeyPathsByPropertyKey {
+    return @{
+             @"displayTitle": @"title",
```

Comment:
@bgerstle When you don't have an instance, the only way to use that macro is to alloc a new instance. Hesitant to to do that here… as this logic can be called frequently. In loops that are out of our control.

Additionally, using this macro in "business logic code" that makes sense… you are referencing key paths while trying to accomplish something else. So the macro is nice because they keep you fro tripping up on keypaths while you are working on other features and keep it in sync with refactors. 

But in this case, the entire point of the code is to define key value mappings - its not ancillary to the task - it is the task. I'm ok with knowing that if someone else or my future self changes properties on a Mantle object, they need to know to update the Key Value mappings. 

Even if someone gets it wrong or is somehow forgotten to update it, then Mantle will break and throw errors/exceptions - immediately when running the app. Additionally, Mapping responses is something we can test with Nocilla.